### PR TITLE
Add "maybe" bindings to SqliteStatement

### PR DESCRIPTION
**Pull request by kabentley**
Allows value to be optional, in which case the method does nothing, leaving the value unchanged (usually `null`).

### CODE REVIEW

File: core/backend/src/SqliteStatement.ts [hunk 1]
```diff
@@ -158,57 +165,117 @@ export class SqliteStatement implements IterableIterator<any>, IDisposable {
...
   public bindInteger(parameter: BindParameter, val: number) {
-    this.checkBind(this.stmt.bindInteger(parameter, val));
+    checkBind(this.stmt.bindInteger(parameter, val));
+  }
+  /** Bind an integer parameter if it is defined. Otherwise do nothing.
+   * @param parameter Index (1-based) or name of the parameter (including the initial ':', '@' or '$')
+   * @param val integer to bind.
+   */
+  public maybeBindInteger(parameter: BindParameter, val?: number) {
+    if (val)
```
Comment (pmconne) [inline core/backend/src/SqliteStatement.ts:180]: What if `val` is `0`?

Comment (kabentley) [reply core/backend/src/SqliteStatement.ts]: yea, i noticed that too

**Review by pmconne — APPROVED** [review]
I am curious what motivated the `maybeBind*` functions.

File: core/backend/src/SqliteStatement.ts (hunk 1 above)
Comment (kabentley) [reply core/backend/src/SqliteStatement.ts]: i'm moving my tests for all of this into this repo. I'll do that next.

**Pull request merged**